python/proton/_message.py

```diff
@@ -446,7 +462,7 @@ def _get_reply_to_group_id(self):
         return utf82unicode(pn_message_get_reply_to_group_id(self._msg))
 
     def _set_reply_to_group_id(self, value):
-        self._check(pn_message_set_reply_to_group_id(self._msg, unicode2utf8(value)))
+        self._check(pn_message_set_reply_to_group_id(self._msg, unicode2utf8(value).encode()))
```

Comment:
I want you to fix the CI issues first. But then, I have also this:

Doing `unicode2utf8(value).encode()` looks cumbersome. First, you convert `value` to `str`, then you convert it to `bytes`. It would make sense to me to replace the `unicode2utf8` usage here with a new function, which would directly return the value converted to `bytes` , and maybe it will also pass through a cffi `char *` values.